multinode_integration_tests/tests/verify_bill_payment_utils/utils.rs

```diff
@@ -0,0 +1,1003 @@
+    let (mut cluster, global_values, _blockchain_server) = establish_test_frame(test_inputs);
+    let consuming_node =
+        global_values.prepare_consuming_node(&mut cluster, &global_values.blockchain_interfaces);
+    let serving_nodes_array = global_values.prepare_serving_nodes(&mut cluster);
+    global_values.set_up_consuming_node_db(&serving_nodes_array, &consuming_node);
+    global_values.set_up_serving_nodes_databases(&serving_nodes_array, &consuming_node);
+    let wholesome_config = WholesomeConfig::new(global_values, consuming_node, serving_nodes_array);
+    wholesome_config.assert_expected_wallet_addresses();
+    let real_consuming_node = cluster.start_named_real_node(
+        &wholesome_config
+            .consuming_node
+            .common
+            .prepared_node
+            .node_docker_name,
+        wholesome_config.consuming_node.common.prepared_node.index,
+        wholesome_config
+            .consuming_node
+            .common
+            .startup_config_opt
+            .borrow_mut()
+            .take()
+            .unwrap(),
+    );
+
+    stimulate_consuming_node_to_pay(&mut cluster, &real_consuming_node, &wholesome_config);
+
+    let timeout_start = Instant::now();
+    while !wholesome_config
+        .consuming_node
+        .payable_dao
+        .non_pending_payables()
+        .is_empty()
+        && timeout_start.elapsed() < Duration::from_secs(10)
+    {
+        thread::sleep(Duration::from_millis(400));
+    }
+    wholesome_config.assert_payments_via_direct_blockchain_scanning(&assertions_values);
+
+    let _ = start_serving_nodes_and_activate_their_accountancy(
+        &mut cluster,
+        // So that individual Configs can be pulled out and used
+        &wholesome_config,
+    );
+
+    wholesome_config.assert_serving_nodes_addressed_received_payments(&assertions_values)
+}
+
+const MNEMONIC_PHRASE: &str =
+    "timber cage wide hawk phone shaft pattern movie army dizzy hen tackle \
+    lamp absent write kind term toddler sphere ripple idle dragon curious hold";
+
+pub struct TestInputs {
+    // The contract owner wallet is populated with 100 ETH as defined in the set of commands with
+    // which we start up the Ganache server.
+    //
+    // This specifies number of wei this account should possess at its initialisation.
+    // The consuming node gets the full balance of the contract owner if left as None. Cannot ever
+    // get more than what the "owner" has.
+    payment_thresholds_all_nodes: PaymentThresholds,
+    node_profiles: NodeProfiles,
+}
+
+#[derive(Default)]
+pub struct TestInputsBuilder {
+    ui_ports_opt: Option<Ports>,
+    consuming_node_initial_tx_fee_balance_minor_opt: Option<u128>,
+    consuming_node_initial_service_fee_balance_minor_opt: Option<u128>,
+    debts_config_opt: Option<DebtsSpecs>,
+    payment_thresholds_all_nodes_opt: Option<PaymentThresholds>,
+    consuming_node_gas_price_opt: Option<u64>,
```

Comment:
You'd like to specify `major` in this field name. The new name will be`consuming_node_gas_price_major_opt`.